File: problems/686/solution.py
class Solution:
    def repeatedStringMatch(self, A: str, B: str) -> int:
        q = (len(B) - 1) // len(A) + 1
        
        for i in range(2):
            if B in A * (q + i):
                return q + i
        
        return -1

File: problems/686/solution.go
package main
import "fmt"
import "strings"

func repeatedStringMatch(A string, B string) int {
    q := (len(B) - 1) / len(A) + 1
    
    fmt.Println(q)
    
    for i := 0; i < 2; i++ {
        if strings.Contains(strings.Repeat(A, q + i), B) {
            return q + i
        }
    }
    
    return -1
}